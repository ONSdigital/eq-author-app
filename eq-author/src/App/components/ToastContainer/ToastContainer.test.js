import React from "react";
import { shallow } from "enzyme";
import { ToastArea, UndoButton } from "./";

describe("Toastcontainer", () => {
  let props;

  const createWrapper = (props, render = shallow) => {
    return render(<ToastArea {...props} />);
  };

  beforeEach(() => {
    const dismissToast = jest.fn();

    const undoToast = jest.fn();

    props = {
      id: "Toast_1",
      toasts: {
        toast1: {
          message: "Hello world",
          undoAction: "undoAction",
          context: {
            sectionId: 1,
            pageId: 1
          }
        }
      },
      dismissToast,
      undoToast
    };
  });

  it("should render", () => {
    const wrapper = createWrapper({ ...props }, shallow);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render the undo button if no undoaction", () => {
    props.toasts.toast1.undoAction = null;
    const wrapper = createWrapper({ ...props }, shallow);
    expect(wrapper.find(UndoButton)).toHaveLength(0);
  });
});
