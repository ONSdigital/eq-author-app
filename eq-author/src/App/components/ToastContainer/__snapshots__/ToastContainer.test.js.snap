// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toastcontainer should render 1`] = `
<ToastContainer__ToastWrapper>
  <ToastContainer__ToastOuterContainer>
    <ToastContainer__ToastInnerContainer>
      <ToastContainer
        transition={[Function]}
      >
        <Toast
          id="toast1"
          key="toast1"
          onClose={[MockFunction]}
          timeout={5000}
        >
          <ToastContainer__StyledDiv>
            Hello world
            <ToastContainer__UndoButton
              data-test="btn-undo"
              onClick={[Function]}
            >
              Undo
            </ToastContainer__UndoButton>
          </ToastContainer__StyledDiv>
        </Toast>
      </ToastContainer>
    </ToastContainer__ToastInnerContainer>
  </ToastContainer__ToastOuterContainer>
</ToastContainer__ToastWrapper>
`;
